EIPS/eip-2016.md

```diff
@@ -0,0 +1,241 @@
+        "symbol": "gorETH"
+      },
+      "blockExplorerUrl": "https://goerli.etherscan.io"
+    }
+  ]
+}
+```
+
+To add POA Network's xDAI chain:
+
+```json
+{
+  "id":1,
+  "jsonrpc": "2.0",
+  "method": "wallet_addEthereumChain",
+  "params": [
+    {
+      "chainId": "0x64",
+      "chainName": "xDAI Chain",
+      "rpcUrl": "https://dai.poa.network",
+      "nativeCurrency": {
+        "name": "xDAI",
+        "symbol": "xDAI",
+        "decimals": 18,
+      }
+    }
+  ]
+}
+```
+
+A success response:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": true
+}
+```
+
+A failure response:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "error": {
+    "code": 4001,
+    "message": "The user rejected the request."
+  }
+}
+```
+
+## Rationale
+
+The design of `wallet_addEthereumChain` is deliberately ignorant of what it means to "add" a chain to a wallet.
+The meaning of "adding" a chain to a wallet depends on the wallet implementation.
+Ultimately, neither the user nor the dapp can now whether adding the chain "worked" until they have successfully interacted with it.
+
+When calling the method, specifying the `chainId` will always be necessary, since in the universe of Ethereum chains, the [EIP-155](./eip-155.md) chain ID is effectively the GUID of a chain.
+The remaining parameters amount to what, in the estimation of authors, a wallet will minimally require in order to effectively support a chain and accurately represent it to the user.
+The network ID (per the `net_version` RPC method) was omitted since it is effectively superseded by the chain ID.
+
+For [security reasons](#security-considerations), a wallet should always attempt to validate the chain metadata provided by the dapp, and may choose to fetch the metadata elsewhere entirely.
+Either way, there's no way to know what the wallet will _need_ with respect to the chain metadata.
+Therefore, all parameters except `chainId` are listed as optional, even though a wallet may require them in practice.
+
+This specification does not mandate that the wallet "switches" its active chain, if the result is successful.
+
+For related work, see [EIP-2015](./eip-2015.md).
+
+For requests of this feature in the community, see e.g. [this GitHub issue](https://github.com/MetaMask/metamask-extension/issues/3604).
+
+## Security Considerations
+
+`wallet_addEthereumChain` is a powerful method that exposes the end user to serious risks if implemented incorrectly.
+Most of these risks can be avoided by validating the request data in the wallet, and clearly disambiguating different chains in the wallet UI.
+
+### Chain IDs
+
+Since the chain ID used for signing determines which chain a transaction is valid for, handling the chain ID correctly is of utmost importance.
+The wallet should:
+
+- Ensure that a submitted chain ID is valid.
+  - It should be submitted as a `0x`-prefixed hexadecimal string per [EIP-695](./eip-695.md), and parse to an integer number.
```

Comment:
This should be in the Rationale, not Security Considerations.

Reply:
I'll just remove it, since the definition of a valid `chainId` is already specified in the "Parameters" section.